### PR TITLE
chore: update sdkv2 and logging sdk deps

### PR DESCRIPTION
**Pull request by cristiGuranIonos**
## Checklist

- [ ] PR name added as appropriate (e.g. `feat:`/`fix:`/`doc:`/`test:`/`refactor:`)
- [ ] Tests added or updated
- [ ] Documentation updated
- [ ] Changelog updated and version incremented (label: upcoming release)
- [ ] Github Issue linked if any
- [ ] Jira task updated

### CODE REVIEW

File: ionoscloud/data_source_vcpu_server.go [hunk 1]
```diff
@@ -10,9 +10,9 @@ func dataSourceVCPUServer() *schema.Resource {
 		ReadContext: dataSourceServerRead,
 		Schema: map[string]*schema.Schema{
 			"datacenter_id": {
-				Type:         schema.TypeString,
-				Required:     true,
-				ValidateFunc: validation.All(validation.StringIsNotWhiteSpace),
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotWhiteSpace),
```
Comment (cbeti-ionos) [inline ionoscloud/data_source_vcpu_server.go:15]: isn't `validation.IsUUID` better suited ?

File: ionoscloud/resource_dataplatform_cluster.go [hunk 2]
```diff
@@ -27,16 +27,17 @@ func resourceDataplatformCluster() *schema.Resource {
 			"datacenter_id": {
 				Type:        schema.TypeString,
 				Description: "The UUID of the virtual data center (VDC) in which the cluster is provisioned",
-				ValidateDiagFunc: validation.ToDiagFunc(validation.All(validation.StringLenBetween(32, 63),
-					validation.StringMatch(regexp.MustCompile("^[0-9a-f]{8}-?[0-9a-f]{4}-?[0-9a-f]{4}-?[0-9a-f]{4}-?[0-9a-f]{12}$"), ""))),
+				ValidateDiagFunc: validation.AllDiag(validation.ToDiagFunc(validation.StringLenBetween(32, 63)),
```
Comment (cbeti-ionos) [inline ionoscloud/resource_dataplatform_cluster.go:30]: can `validation.IsUUID` be used here instead ?

Comment (cristiGuranIonos) [reply ionoscloud/resource_dataplatform_cluster.go]: done

File: ionoscloud/resource_dataplatform_cluster.go [hunk 3]
```diff
@@ -27,16 +27,17 @@ func resourceDataplatformCluster() *schema.Resource {
...
 				ForceNew: true,
 			},
 			"name": {
-				Type:             schema.TypeString,
-				Description:      "The name of your cluster. Must be 63 characters or less and must be empty or begin and end with an alphanumeric character ([a-z0-9A-Z]). It can contain dashes (-), underscores (_), dots (.), and alphanumerics in-between.",
-				ValidateDiagFunc: validation.ToDiagFunc(validation.All(validation.StringLenBetween(0, 63), validation.StringMatch(regexp.MustCompile("^[A-Za-z0-9][-A-Za-z0-9_.]*[A-Za-z0-9]$"), ""))),
-				Required:         true,
+				Type:        schema.TypeString,
+				Description: "The name of your cluster. Must be 63 characters or less and must be empty or begin and end with an alphanumeric character ([a-z0-9A-Z]). It can contain dashes (-), underscores (_), dots (.), and alphanumerics in-between.",
+				ValidateDiagFunc: validation.AllDiag(validation.ToDiagFunc(validation.StringLenBetween(0, 63)),
```
Comment (cbeti-ionos) [inline ionoscloud/resource_dataplatform_cluster.go:37]: Could make a custom validation `VerifyAlphanumericString` or similar which also takes a length parameter and maybe a separator slice for the `_ - .` to merge these two into. 
This is already used in 2 different places, and might be used even more in the future.

Comment (cristiGuranIonos) [reply ionoscloud/resource_dataplatform_cluster.go:38]: good idea, but not necessary for this case

**Review by cbeti-ionos — APPROVED** [review]

**Pull request merged**